### PR TITLE
[SG-487] Web component unit tests

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- Add Jest to Web
- Add unit tests to a discreet component to use as an example (I picked TrialInitiation component because that was some recent new development).

## Code changes

Changes needed to enable jest:
- **apps/web/package.json:** Add test scripts to package.json
- **apps/web/jest.config.js:** Add jest config
- **apps/web/test.config.ts:** Add test config
- **apps/web/tsconfig.spec.json:** add spect tsconfig
- **apps/web/.eslintrc.json:** This change wasn't needed to enable jest per se, but it does fix an issue with importing testing classes from angular/core

Added tests:
- **apps/web/src/app/modules/trial-initiation/trial-initiation.component.spec.ts:** Added 13 tests for the Trial Initiation component that demonstarte a broad set of testing conditions, including:
    - Mocking service calls
    - Route params
    - Router navigation
    - The use of a stub component
- **apps/web/src/app/modules/trial-initiation/trial-initiation.component.ts:** Small fix I found from testing 😁 

## Screenshots
Nothing fancy to demonstrate, just some passing tests 👩‍🍳 🤌 

Screen Shot 2022-08-03 at 3 07 34 PM

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [x] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/web/.eslintrc.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "no-restricted-imports": [
       "error",
       {
-        "patterns": ["**/core/*"]
+        "patterns": ["**/app/core/*"]
```
Comment (differsthecat) [inline apps/web/.eslintrc.json:9]: @Hinton I see you added this when adding the CoreModule in [this PR](https://github.com/bitwarden/clients/pull/3149/files), but it is preventing me from adding imports from `"@angular/core/testing"`. I'm sure this is not the correct update, I added it in order to allow me to commit the test changes.

Comment (Hinton) [reply apps/web/.eslintrc.json:9]: Yea that wasn't intended.

**Comment by mimartin12** [comment]
@bitwarden/dept-devops we should look at adding these into the web CI workflows as a `Test` step. Great work @differsthecat  👍

File: apps/web/src/app/modules/trial-initiation/trial-initiation.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+import { VerticalStepperComponent } from "../vertical-stepper/vertical-stepper.component";
+
+import { TrialInitiationComponent } from "./trial-initiation.component";
+
+describe("TrialInitiationComponent", () => {
+  let component: TrialInitiationComponent;
+  let fixture: ComponentFixture<TrialInitiationComponent>;
+  const mockQueryParams = new BehaviorSubject<any>({ org: "enterprise" });
+  const testOrgId = "91329456-5b9f-44b3-9279-6bb9ee6a0974";
```
Comment (addisonbeck) [inline apps/web/src/app/modules/trial-initiation/trial-initiation.component.spec.ts:28]: Is this GUID significant, or just a random value? Would using [crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) work?

Comment (differsthecat) [reply apps/web/src/app/modules/trial-initiation/trial-initiation.component.spec.ts:28]: It's just a random value and has no significance. I like your suggestion to generate it!

Comment (differsthecat) [reply apps/web/src/app/modules/trial-initiation/trial-initiation.component.spec.ts:28]: @addisonbeck I played around with this a bit, but unfortunately `crypto` isn't available in the jest context. I can import a library to fix this, but that seems like overkill.

Comment (addisonbeck) [reply apps/web/src/app/modules/trial-initiation/trial-initiation.component.spec.ts:28]: I agree

**Review by addisonbeck — APPROVED** [review]

**Pull request merged**